src/expression.d

```diff
@@ -14159,14 +14159,19 @@ extern (C++) final class MinExp : BinExp
         {
             if (t2.ty == Tpointer)
             {
+                Type p1 = t1.nextOf();
+                Type p2 = t2.nextOf();
+
+                if (!p1.equivalent(p2))
```

Comment:
Phobos and druntime have been mostly chill as all the errors were due to the use of `void*` and `byte*`. Relying on the size of the types still allows the user to mix and match the types which is not what you want in 99% of the cases (and if you really need to do so you're just explicit cast away) as some types may vary between platforms and one may end up enlarging/shrinking one of the types involved and end up with a not so nice compilation error.
I root for following the specification here.